src/cdp/connection.ts

```diff
@@ -0,0 +1,116 @@
+import { IServer } from '../utils/iServer';
+import { CdpMessage } from './message';
+import { CdpClient, createClient } from './cdpClient';
+
+import { log } from '../utils/log';
+const logCdp = log('cdp');
+
+interface CdpCallbacks {
+  resolve: (messageObj: {}) => void;
+  reject: (errorObj: {}) => void;
+}
+
+/**
+ * Represents a high-level CDP connection to the browser backend.
+ * Manages a CdpClient instance for each active CDP session.
+ */
+export class Connection {
+  private _browserClient: CdpClient;
```

Comment:
I'd prefer to use the name _browserCdpClient here since this CDP target is known as the "browser target" in CDP terminology. See for example `Target.attachToBrowserTarget`.